src/interaction/agent_env_interaction.py

```diff
@@ -0,0 +1,26 @@
+"""Contains classes for handling the interaction between an agent and
```

Comment:
is this all this file is ever going to be? if so, move to structs?

Reply:
i thought it would be fancier when i started.... done